docs/recipes/endpoint-testing-with-mongoose.md

```diff
@@ -0,0 +1,281 @@
+# Endpoint testing with Mongoose
+
+This recipe shows you how to test your endpoints with AVA and Mongoose.
+
+## Setup
+
+This recipe uses the following libraries:
+
+1. [MongoDB memory server](https://github.com/nodkz/mongodb-memory-server) (A MongoDB in-memory Server)
+2. [Babel Polyfill](https://babeljs.io/docs/usage/polyfill/) (required for MongoDB memory server)
+3. [Supertest](https://github.com/visionmedia/supertest) (An endpoint testing library)
+4. Mongoose
+
+Install the first three libraries by running the following code:
+
+```console
+$ npm install --save-dev mongodb-memory-server babel-polyfill supertest
+```
+
+You should have Mongoose installed already. If not, run the following code to install it:
+
+(Note: You need at least mongoose v4.11.3)
+
+```console
+$ npm install mongoose
+```
+
+## Setting up AVA
+
+Since MongoDB Memory server requires Babel polyfill to work, the easiest way to set up AVA is through the `ava` key in your `package.json` file.
```

Comment:
I tried adding regenerator-runtime, but it didn't work. Looks like regenerator-runtime is required in mongodb-memory-server? 

Here's the stack trace: 

```
/Users/zellwk/Projects/test-repos/ava-mdb-test/node_modules/mongodb-memory-server/lib/MongoMemoryServer.js:10
  var _ref = _asyncToGenerator(regeneratorRuntime.mark(function _callee(port, dbName) {
                               ^

ReferenceError: regeneratorRuntime is not defined
    at /Users/zellwk/Projects/test-repos/ava-mdb-test/node_modules/mongodb-memory-server/lib/MongoMemoryServer.js:10:32
    at Object.<anonymous> (/Users/zellwk/Projects/test-repos/ava-mdb-test/node_modules/mongodb-memory-server/lib/MongoMemoryServer.js:44:2)
```